### PR TITLE
libgrape-lite v0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by sighingnow**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by sighingnow** [comment]
Could someone help to get this PR landed?

Thanks!

File: Formula/libgrape-lite.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  desc "C++ library for parallel graph processing"
+  homepage "https://github.com/alibaba/libgrape-lite"
+  url "https://github.com/alibaba/libgrape-lite/archive/refs/tags/v0.1.0.tar.gz"
+  sha256 "aafc8e5e9b122c4bde4370ee3f2ab8afb043f2a1547cacc062991ad420e3b9a4"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+
+  depends_on "glog"
+  depends_on "open-mpi"
```
Comment (SMillerDev) [inline Formula/libgrape-lite.rb:11]: Does it not work with the macOS version?

File: Formula/libgrape-lite.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  on_linux do
+    depends_on "gcc"
```
Comment (SMillerDev) [inline Formula/libgrape-lite.rb:14]: Does GCC 5 not work?

File: Formula/libgrape-lite.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  end
+
+  def install
+    ENV.cxx11
+
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_CXX_STANDARD=11",
```
Comment (SMillerDev) [inline Formula/libgrape-lite.rb:21]: Is this needed along the ENV.cxx11?

File: Formula/libgrape-lite.rb (hunk 1 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb:11]: It does work on macOS. `open-mpi` is available on Mac in `brew install` as well. Could you please explain a bit more about the concern about macOS here?

File: Formula/libgrape-lite.rb (hunk 2 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb:14]: I was assuming `depends_on "gcc"` will bring the default gcc. I will delete this part.

File: Formula/libgrape-lite.rb (hunk 3 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb:21]: Building this library requires c++11. Doesn't `Env.cxx11` means c++11?

File: Formula/libgrape-lite.rb (hunk 2 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb]: Fixed.

File: Formula/libgrape-lite.rb (hunk 3 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb:15]: I have delete the `ENV.cxx11` line.

Comment (SMillerDev) [reply Formula/libgrape-lite.rb:15]: Yes, but do you still need to tell CMake separately?

File: Formula/libgrape-lite.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/libgrape-lite.rb:11]: As far as I remember macOS comes with an open-mpi alternative. Can we not use that?

Comment (sighingnow) [reply Formula/libgrape-lite.rb:11]: I haven't found a builtin MPI on my Mac, but I found the following statement from the open-mpi FAQ: https://www.open-mpi.org/faq/?category=osx

> Open MPI v1.2.3 was included in some older versions of OS X, starting with version 10.5 (Leopard). It was removed in more recent versions of OS X (we're not sure in which version it disappeared — *but your best bet is to simply download a modern version of Open MPI for your modern version of OS X*).
> 
> Note, however, that OS X Leopard does not include a Fortran compiler, so the OS X-shipped version of Open MPI does not include Fortran support.
> 
> If you need/want Fortran support, you will need to build your own copy of Open MPI (assumedly when you have a Fortran compiler installed). The Open MPI team strongly recommends not overwriting the OS X-installed version of Open MPI, but rather installing it somewhere else (e.g., /opt/openmpi).


I think it was removed in recently versions of Mac OS (may as early as 10.15, the oldest version we are running the CI). So I think it should be OK to install the "open-mpi" package using the homebrew as the dependency ?

File: Formula/libgrape-lite.rb (hunk 3 above)
Comment (sighingnow) [reply Formula/libgrape-lite.rb:15]: I would use `ENV.cxx11` and remove the `-DDCMAKE_CXX_STANDARD=11` lines.

Comment (sighingnow) [reply Formula/libgrape-lite.rb]: Fixed.

File: Formula/libgrape-lite.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/libgrape-lite.rb:11]: I guess I was wrong then.

**Review by SMillerDev — APPROVED** [review]
Thanks @sighingnow  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**